infrastructure-formula/infrastructure/salt/proxy_networkautomation.sls

```diff
@@ -0,0 +1,135 @@
+{%- set mypillar = salt['pillar.get']('infrastructure:salt:proxy', {}) -%}
+
+{%- if mypillar %}
+salt_pod_user:
+  group.present:
+    - name: autopod
+    - system: True
+    - gid: 900
+
+  user.present:
+    - name: autopod
+    - system: True
+    - home: /var/lib/autopod
+    - uid: 900
+    - gid: 900
+    - require:
+      - group: autopod
+    - require_in:
+      - User session for autopod is initialized at boot
+
+  cmd.run:
+    - name: 'usermod --add-subuids 100000-165535 --add-subgids 100000-165535 autopod'
+    - onchanges:
+      - group: autopod
+      - user: autopod
+    - require_in:
+      - User session for autopod is initialized at boot
+
+salt_pod_dirs:
+  file.directory:
+    - makedirs: True
+    - user: root
+    - group: autopod
+    - mode: '0750'
+    - names:
+      - /etc/salt-pod/pki/minion
+      - /etc/salt-pod/pki/proxy
+      - /etc/salt-pod/proxy.d
+    - require:
+      - user: salt_pod_user
+
+salt_podpki_files:
+  file.managed:
+    - user: root
+    - group: autopod
+    - names:
+      {%- for mode in ['minion', 'proxy'] %}
+      - /etc/salt-pod/pki/{{ mode }}/minion_master.pub:
+        - contents_pillar: 'infrastructure:salt:proxy:podpki:master'
+        - mode: '0644'
+      - /etc/salt-pod/pki/{{ mode }}/minion.pub:
+        - contents_pillar: 'infrastructure:salt:proxy:podpki:crt'
+        - mode: '0644'
+      - /etc/salt-pod/pki/{{ mode }}/minion.pem:
+        - contents_pillar: 'infrastructure:salt:proxy:podpki:key'
+        - mode: '0440'
+      {%- endfor %}
+      - /etc/motd:
+        - contents:
+          - This machine runs multiple Salt minion and proxy instances in individual containers.
+          - Use `systemctl -Mautopod@ --user <start|stop|restart|status> [pattern]` to manage containers.
+          - Use `journalctl [-t pattern]` to inspect container logs.
+          - Do not modify Podman containers under the autopod@ user manually, they are managed by Salt.
+    - require:
+      - salt_pod_user
+      - file: salt_pod_dirs
+
+salt_pod_proxy_config:
+  file.managed:
+    - names:
+      - /etc/salt-pod/proxy:
+        - contents: |
+            # Managed by Salt
+            {%- if 'master' in mypillar %}
+            master:
+              {%- for master in mypillar['master'] %}
+              - {{ master }}
+              {%- endfor %}
+            {%- endif %}
+            log_level: info
+            saltenv: production_network
```

Comment:
Another hardcoded branch name

Reply:
Same as with https://github.com/openSUSE/salt-formulas/pull/49#discussion_r1296082551, it's configured like

```
salt:
  master:
    env_order:
      - devel
      - devel_network
    {%- if 'salt.proxy_master' in roles %}
    pillar_roots:
      __env__:
        - /srv/pillar_network
    {%- endif %}
    ext_pillar_first: True
    fileserver_backend:
      - roots
      - git
    file_roots:
      devel_network:
        - /usr/share/salt-formulas/states
```

And `/srv/pillar_network` is populated by a CI script. So, the name is in use in multiple places and I'm not sure moving one of them to a variable is helpful.

Reply:
Same story, though I do think some logic to centrally change it could be useful for a non-production environment. Currently I maintain a separate copy of the files. It could be a Jinja variable in the pillar which then propagates to the state.

Reply:
I added https://github.com/openSUSE/salt-formulas/issues/52 as a reminder for myself.